### PR TITLE
Simplify the mapping config for QGIS server

### PR DESCRIPTION
**Pull request by sbrunner**
Fix GSGMF-1605

### CODE REVIEW

File: docker/qgisserver/geomapfish_qgisserver/accesscontrol.py [hunk 1]
```diff
@@ -110,10 +110,68 @@ def __init__(self, server_iface: qgis.server.QgsServerInterface):
...
+                        if (
+                            base_url.schema == url.schema
+                            and base_url.netloc == url.netloc
+                            and base_url.path == url.path
+                        ):
+                            query = url.query_lower
+                            if "map" not in query:
+                                LOG.error(
+                                    "Ignoring OGC server '%s', 'map' is not in the parameters", ogcserver.name
+                                )
```
Comment (sbrunner) [inline docker/qgisserver/geomapfish_qgisserver/accesscontrol.py:150]: Should try single mode

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/docker-compose.yaml [hunk 2]
```diff
@@ -32,12 +32,14 @@ services:
   ##   volumes_from:
   ##     - config:ro
   ##   environment:
+  ##     # Auto
+  ##     - QGISSERVER_URL
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/docker-compose.yaml:36]: Should be defined ...

File: doc/integrator/backend_qgis.rst [hunk 3]
```diff
@@ -177,23 +177,9 @@ The access restriction is available only for Docker projects.
...
-     <project file path>:
-       ogc_server: <OGC server name>
-
-``<project file path>`` should have exactly the same value as the ``MAP`` parameter in the ``Base URL``
-value of the OGC server.
-
-Finally, you should provide the ``GEOMAPFISH_ACCESSCONTROL_CONFIG`` to point to a config file e.-g.
-``/etc/qgisserver/accesscontrol_config.yaml``, and ``QGIS_PROJECT_FILE`` to be empty.
+From version 2.7 the config is just made with the ``GEOMAPFISH_ACCESSCONTROL_BASE_URL`` environment
+variable the contanis the the docker url present in the OGC servers, by default it's et to
```
Comment (arnaud-morvan) [inline doc/integrator/backend_qgis.rst:181]:
```suggestion
variable which contains the base URL of the OGC servers, by default it's set to
```

File: doc/integrator/backend_qgis.rst [hunk 4]
```diff
@@ -177,23 +177,9 @@ The access restriction is available only for Docker projects.
...
+``QGISSERVER_URL``. And the plugin wikk search the OGC servers that corresponds to this URL.
```
Comment (arnaud-morvan) [inline doc/integrator/backend_qgis.rst:182]:
```suggestion
``QGISSERVER_URL``. And the plugin will search for the OGC servers that match with this base URL.
```

File: docker/qgisserver/geomapfish_qgisserver/accesscontrol.py [hunk 5]
```diff
@@ -110,10 +111,103 @@ def __init__(self, server_iface: qgis.server.QgsServerInterface):
...
+                            if "map" not in query:
+                                if single_ogc_server is None:
+                                    single_ogc_server = ogcserver
+                                    LOG.debug(
+                                        "OGC server '%s', 'map' is not in the parameters => single server?",
+                                        ogcserver.name,
+                                    )
+                                else:
+                                    LOG.error(
+                                        "OGC server '%s', 'map' is not in the parameters and we already have a singler OCG server '%s'",
```
Comment (arnaud-morvan) [inline docker/qgisserver/geomapfish_qgisserver/accesscontrol.py:157]:
```suggestion
                                        "OGC server '%s', 'map' is not in the parameters and we already have a single OCG server '%s'",
```

File: docker/qgisserver/geomapfish_qgisserver/accesscontrol.py [hunk 6]
```diff
@@ -110,10 +111,103 @@ def __init__(self, server_iface: qgis.server.QgsServerInterface):
...
+                        else:
+                            LOG.debug(
+                                "Ignoring OGC server '%s', Don't match the base URL '%s' and '%s'",
+                                ogcserver.name,
+                                base_url,
+                                url,
+                            )
+                    if self.ogcserver_accesscontrols and single_ogc_server is not None:
+                        if os.environ.get("QGIS_PROJECT_FILE"):
+                            LOG.error("We have single and multiy ogc server, using single...")
```
Comment (arnaud-morvan) [inline docker/qgisserver/geomapfish_qgisserver/accesscontrol.py:185]:
```suggestion
                            LOG.error("We have OGC servers with and without parameter MAP and a value in QGIS_PROJECT_FILE, fallback to single OGC server mode.")
```

File: docker/qgisserver/geomapfish_qgisserver/accesscontrol.py [hunk 7]
```diff
@@ -110,10 +111,103 @@ def __init__(self, server_iface: qgis.server.QgsServerInterface):
...
+                            self.ogcserver_accesscontrols = {}
+                        else:
+                            LOG.error("We have single and multiy ogc server, using multi...")
```
Comment (arnaud-morvan) [inline docker/qgisserver/geomapfish_qgisserver/accesscontrol.py:188]:
```suggestion
                            LOG.error("We have OGC servers with and without parameter MAP but no value in QGIS_PROJECT_FILE, fallback to mutiple OGC server mode.")
```

File: doc/integrator/backend_qgis.rst [hunk 8]
```diff
@@ -177,23 +177,9 @@ The access restriction is available only for Docker projects.
...
-   map_config:
-     <project file path>:
-       ogc_server: <OGC server name>
-
-``<project file path>`` should have exactly the same value as the ``MAP`` parameter in the ``Base URL``
-value of the OGC server.
-
-Finally, you should provide the ``GEOMAPFISH_ACCESSCONTROL_CONFIG`` to point to a config file e.-g.
-``/etc/qgisserver/accesscontrol_config.yaml``, and ``QGIS_PROJECT_FILE`` to be empty.
+From version 2.7 the config is just made with the ``GEOMAPFISH_ACCESSCONTROL_BASE_URL`` environment
```
Comment (arnaud-morvan) [inline doc/integrator/backend_qgis.rst:180]: Seems that the others modes are still supported, but documentation already removed.

File: geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 9]
```diff
@@ -1,6 +1,23 @@
...
+
+Information to know before starting the upgrade
+===============================================
+
+Information
+===========
+
+1. The congiguration of the QGIS authentication module is simplified in standard case the configuration
+   fie  (`qgisserver/accesscontrol_config.yaml`) and the specified configuration in the `docker-compose.yaml`
+   file can be removed.
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl:16]: This sentence is hard to understand, seems two things are mixed together.

What is simplified in QGIS authentication module ?

You do not talk about GEOMAPFISH_ACCESSCONTROL_BASE_URL ?

File: docker/qgisserver/tests/functional/accesscontrol_test.py [hunk 10]
```diff
@@ -512,3 +514,38 @@ def test_init(self, server_iface, test_data):
...
             plugin.get_ogcserver_accesscontrol()
+
+
+@pytest.mark.usefixtures(
+    "qgs_access_control_filter",
+    "auto_multi_ogc_server_env",
+    "test_data",
+)
+class TestGeoMapFishAccessControlAutoMultiOGCServer:
+    @pytest.mark.usefixtures()
```
Comment (arnaud-morvan) [inline docker/qgisserver/tests/functional/accesscontrol_test.py:525]: This seems to have no effect.

File: docker/qgisserver/tests/functional/accesscontrol_test.py [hunk 11]
```diff
@@ -512,3 +514,38 @@ def test_init(self, server_iface, test_data):
...
+        assert plugin.get_ogcserver_accesscontrol().ogcserver.name == "qgisserver1"
+
+
+@pytest.mark.usefixtures(
+    "qgs_access_control_filter",
+    "auto_single_ogc_server_env",
+    "test_data",
+)
+class TestGeoMapFishAccessControlAutoSingleOGCServer:
+    @pytest.mark.usefixtures()
```
Comment (arnaud-morvan) [inline docker/qgisserver/tests/functional/accesscontrol_test.py:543]: This seems to have no effect.

File: docker/qgisserver/tests/functional/conftest.py [hunk 12]
```diff
@@ -97,3 +97,23 @@ def multiple_ogc_server_env():
...
+
+
+@pytest.fixture(scope="class")
+def auto_multi_ogc_server_env():
+    os.environ["GEOMAPFISH_ACCESSCONTROL_BASE_URL"] = "http://qgis"
+    old = os.environ["QGIS_PROJECT_FILE"]
+    del os.environ["QGIS_PROJECT_FILE"]
+    yield
+    del os.environ["GEOMAPFISH_ACCESSCONTROL_BASE_URL"]
+    os.environ["QGIS_PROJECT_FILE"] = old
```
Comment (arnaud-morvan) [inline docker/qgisserver/tests/functional/conftest.py:109]: Could be:
```
def auto_multi_ogc_server_env()
    with patch.dict(
        "geomapfish_qgisserver.accesscontrol.os.environ",
        {
            "GEOMAPFISH_ACCESSCONTROL_BASE_URL": "http://qgis",
            ...
        }
    ):
        yield
```
See: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict

File: doc/integrator/backend_qgis.rst (hunk 8 above)
Comment (sbrunner) [reply doc/integrator/backend_qgis.rst:180]: Yes, you prefer that I put them is a deprecated section?

File: geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 9 above)
Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Done...

File: docker/qgisserver/tests/functional/conftest.py (hunk 12 above)
Comment (sbrunner) [reply docker/qgisserver/tests/functional/conftest.py]: Thanks :-)

File: geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 13]
```diff
@@ -1,6 +1,25 @@
...
+Version 2.7.0
+=============
+
+Information to know before starting the upgrade
+===============================================
+
+Information
+===========
+
+1. The congiguration of the QGIS authentication module is simplified,
```
Comment (jwkaltz) [inline geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl:14]: congiguration -> configuration

File: geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 14]
```diff
@@ -1,6 +1,25 @@
...
+   now a new environment variable `GEOMAPFISH_ACCESSCONTROL_BASE_URL` is set to the base URL of
+   qgisserver, and we search for all the OGC server that corresponds to this URL.
```
Comment (jwkaltz) [inline geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl:16]: server -> servers
corresponds -> correspond

**Review by jwkaltz — COMMENTED** [review]
Could we rename the PR to "Simplify the mapping config for QGIS server"?

**Review by arnaud-morvan — APPROVED** [review]

**Pull request merged**